src/FlatSelect/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,222 @@
+	private static DropdownIdentifier: number = 0;
+
+	public state = {
+		dropdownId: "",
+		isInBottomOfScreen: false,
+		show: false
+	};
+
+	private wasEventListenerAttached: boolean = false;
+	private panelRef = React.createRef<Panel>();
+	private dropdownRef = React.createRef<HTMLDivElement>();
+
+	public hide = () => {
+		this.changeVisibility(false);
+	};
+
+	public componentWillMount() {
+		Dropdown.DropdownIdentifier++;
+		this.setState(state => {
+			const newState = {
+				...state,
+				dropdownId: "react-dropdown-" + Dropdown.DropdownIdentifier
+			};
+
+			const dropdownPanelContainer = document.createElement("div");
+			dropdownPanelContainer.id = newState.dropdownId;
+			const overlay = Dropdown.getOverlay();
+			overlay.appendChild(dropdownPanelContainer);
+
+			return newState;
+		});
+	}
+
+	public componentWillUnmount() {
+		const dropdownPanel = document.getElementById(this.state.dropdownId);
+
+		if (dropdownPanel) {
+			Dropdown.getOverlay().removeChild(dropdownPanel);
+			unmountComponentAtNode(dropdownPanel);
+		}
+	}
+
+	public componentDidUpdate() {
+		this.renderPanel();
+	}
+
+	public componentDidMount() {
+		this.renderPanel();
+	}
+
+	public render() {
+		const classesArray = [
+			this.props.className || "",
+			"form-control",
+			"kit-selectR",
+			"kit-selectR-container"
+		];
+
+		if (this.state.show) {
+			classesArray.push("kit-selectR-open");
+			classesArray.push(this.props.openedClassName || "");
+
+			if (this.state.isInBottomOfScreen) {
+				classesArray.push("kit-selectR-above");
+			}
+		} else {
+			classesArray.push(this.props.closedClassName || "");
+		}
+
+		if (this.props.height) {
+			classesArray.push(String(Height.getClass(this.props.height)));
+		}
+
+		if (this.props.width) {
+			classesArray.push(String(Width.getClass(this.props.width)));
+		}
+
+		if (!this.props.headerInfo) {
+			classesArray.push("kit-selectR-placeholder");
+		}
+
+		if (this.props.disabled) {
+			classesArray.push("kit-selectR-disabled");
+		}
+
+		const classes = classesArray.reduce((curr, next) => curr + " " + next);
+
+		const placeholder = this.props.headerInfo ? (
+			<span className="kit-selectR-chosen">{this.props.headerInfo}</span>
+		) : (
+			<span className="kit-selectR-chosen">{this.props.placeholder}</span>
+		);
+
+		const style = { ...this.props.style, marginLeft: "0 !important" };
+		return (
+			<div
+				id={this.props.id}
+				className={classes}
+				onClick={() => this.changeVisibility()}
+				style={style}
+				ref={this.dropdownRef}
+			>
+				<span className="kit-selectR-choice">
+					{placeholder}
+					{this.clearSelectionSection()}
+				</span>
+			</div>
+		);
+	}
+
+	private clearSelectionSection = (): JSX.Element | null => {
+		return this.props.onSelectionClear == null ? null : (
+			<abbr
+				className="select2-search-choice-close"
+				onClick={this.onSelectionClear}
+			/>
+		);
+	};
+
+	private onSelectionClear = (e: React.SyntheticEvent<{}>) => {
```

Comment:
Почему не MouseEvent?
```suggestion
	private onSelectionClear = (e: React.MouseEvent) => {
```